cognite/neat/_graph/transformers/_prune_graph.py

```diff
@@ -1,84 +1,131 @@
         self.delete_connecting_node = delete_connecting_node
+        self.predicate_property_name = predicate_property_name
 
-    def transform(self, graph: Graph) -> None:
-        nodes_to_delete: list[Identifier] = []
-
-        graph_traversals = list(
-            graph.query(
-                self._query_template.format(
-                    destination_node_type=self.destination_node_type,
-                    property_predicate=self.property_predicate,
-                    property_name=self.property_name,
-                )
+    def transform(self, graph) -> None:
+        nodes_to_delete: list = []
+
+        if self.predicate_property_name is not None:
+            predicate_property_predicate = self.namespace[self.predicate_property_name]
+            query = self._query_template_new_predicate.format(
+                destination_node_type=self.destination_node_type,
+                predicate_property_predicate=predicate_property_predicate,
+                value_property_predicate=self.value_property_predicate,
+            )
+        else:
+            query = self._query_template_keep_old_predicate.format(
+                destination_node_type=self.destination_node_type,
+                value_property_predicate=self.value_property_predicate,
+            )
+
+        graph_traversals = graph.query(query)
+
+        for result_raw in graph_traversals:
+            source_node, old_predicate, destination_node, new_predicate_value, new_property_value = (
+                result_raw.asdict().values()
             )
-        )
-
-        for path in graph_traversals:
-            if isinstance(path, ResultRow):
-                source_node, predicate, destination_node, property_value = path.asdict().values()
-
-                # Create new connection from source node to value
-                graph.add((source_node, predicate, property_value))
-                nodes_to_delete.append(destination_node)
-
-            if self.delete_connecting_node:
-                for node in nodes_to_delete:
-                    # Remove edge triples to node
-                    graph.remove((None, None, node))
-                    # Remove node triple
-                    graph.remove((node, None, None))
+
+            if self.predicate_property_name is not None:
+                # Ensure new predicate is URI compliant as we are creating a new predicate
+                new_predicate_value_string = sentence_or_string_to_camel(str(new_predicate_value))
+                predicate = as_neat_compliant_uri(self.namespace[new_predicate_value_string])
+            else:
+                predicate = old_predicate
+
+            # Create new connection from source node to value
+            graph.add((source_node, predicate, new_property_value))
+            # Remove old relationship between source node and destination node
+            graph.remove((source_node, predicate, destination_node))
+
+            nodes_to_delete.append(destination_node)
```

Comment:
you need to store both source and destination node
```suggestion
            nodes_to_delete.append((source_node, destination_node))
```

Reply:
I am already deleting that triple in row 119 though. So I just want to collect the TargetNode to remove it if the `delete_connecting_node` flag is set.